python/ray/autoscaler/_private/local/config.py

```diff
@@ -1,2 +1,94 @@
-def bootstrap_local(config):
+import copy
+from typing import Any
+from typing import Dict
+
+from ray.autoscaler._private.cli_logger import cli_logger
+
+unsupported_field_message = ("The field {} is not supported "
+                             "for on-premise clusters.")
+
+NODE_TYPE = "ray.node"
+
+
+def prepare_local(config: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Prepare local cluster config for ingestion by cluster launcher and
+    autoscaler.
+    """
+    config = copy.deepcopy(config)
+    for field in "head_node", "worker_nodes", "available_node_types":
+        if config.get(field):
+            err_msg = unsupported_field_message.format(field)
+            cli_logger.abort(err_msg)
+    config["available_node_types"] = {
+        NODE_TYPE: {
+            "node_config": {},
+            "resources": {}
+        }
+    }
+    config["head_node_type"] = NODE_TYPE
+    if "coordinator_address" in config["provider"]:
+        config = prepare_coordinator(config)
+    else:
+        config = prepare_manual(config)
+    return config
+
+
+def prepare_coordinator(config: Dict[str, Any]) -> Dict[str, Any]:
+    config = copy.deepcopy(config)
+    # User should explicitly set the max number of workers for the coordinator
+    # to allocate.
+    if "max_workers" not in config:
+        cli_logger.abort("The field `max_workers` is required when using an "
+                         "automatically managed on-premise cluster.")
+    node_type = config["available_node_types"][NODE_TYPE]
+    node_type["min_workers"] = config.pop("min_workers", 0)
```

Comment:
```suggestion
	# Global `min_workers` are no longer used.
    node_type["min_workers"] = config.pop("min_workers", 0)
```

Or something like this!